ZipBuilder/Sources/ZipBuilder/CocoaPodUtils.swift

```diff
@@ -148,31 +117,65 @@ public enum CocoaPodUtils {
     }
   }
 
-  /// Load versions of installed Pods from the contents of a `Podfile.lock` file.
+  /// Load installed Pods from the contents of a `Podfile.lock` file.
   ///
   /// - Parameter contents: The contents of a `Podfile.lock` file.
-  /// - Returns: A dictionary with names of the pod for keys and a string representation of the
-  ///            version for values.
-  public static func loadVersionsFromPodfileLock(contents: String) -> [String: String] {
-    // This pattern matches a framework name with its version (two to three components)
+  /// - Returns: An array of PodInfo structs.
+  public static func loadPodInfoFromPodfileLock(contents: String) -> [String: PodInfo] {
+    // This pattern matches a pod name with its version (two to three components)
     // Examples:
     //  - FirebaseUI/Google (4.1.1):
     //  - GoogleSignIn (4.0.2):
 
     // Force unwrap the regular expression since we know it will work, it's a constant being passed
     // in. If any changes are made, be sure to run this script to ensure it works.
-    let regex = try! NSRegularExpression(pattern: " - (.+) \\((\\d+\\.\\d+\\.?\\d*)\\)",
-                                         options: [])
+    let podRegex = try! NSRegularExpression(pattern: " - (.+) \\((\\d+\\.\\d+\\.?\\d*)\\)",
+                                            options: [])
+    let depRegex: NSRegularExpression = try! NSRegularExpression(pattern: " - (.+).*",
+                                                                 options: [])
     let quotes = CharacterSet(charactersIn: "\"")
-    var frameworks: [String: String] = [:]
-    contents.components(separatedBy: .newlines).forEach { line in
-      if let (framework, version) = detectVersion(fromLine: line, matching: regex) {
-        let coreFramework = framework.components(separatedBy: "/")[0]
-        let key = coreFramework.trimmingCharacters(in: quotes)
-        frameworks[key] = version
+    var pods: [String: String] = [:]
+    var deps: [String: Set<String>] = [:]
+    var currentPod: String?
+    for line in contents.components(separatedBy: .newlines) {
+      if line.starts(with: "DEPENDENCIES:") {
+        break
+      }
+      if let (pod, version) = detectVersion(fromLine: line, matching: podRegex) {
+        let corePod = pod.components(separatedBy: "/")[0]
+        currentPod = corePod.trimmingCharacters(in: quotes)
+        pods[currentPod!] = version
+      } else if let curPod = currentPod {
```

Comment:
Nit: I believe we should be able to do `else if let currentPod = currentPod` which gives you an unwrapped version of the same name, it makes it a bit more readable I find.

Reply:
Thanks!